src/cpp/topic/Topic.cpp

```diff
@@ -19,40 +19,49 @@
 namespace eprosima {
 namespace uxr {
 
-Topic::Topic(const dds::xrce::ObjectId& object_id, Middleware* middleware, const std::shared_ptr<Participant>& participant)
-    : XRCEObject(object_id, middleware),
-      participant_(participant)
+Topic* Topic::create(
+        const dds::xrce::ObjectId& object_id,
+        const std::shared_ptr<Participant>& participant,
+        const dds::xrce::OBJK_TOPIC_Representation& representation)
 {
-    participant_->tie_object(object_id);
-}
-
-Topic::~Topic()
-{
-    participant_->untie_object(get_id());
-    middleware_->delete_topic(get_raw_id(), participant_->get_raw_id());
-}
+    bool created_entity = false;
+    uint16_t raw_object_id = uint16_t((object_id[0] << 8) + object_id[1]);
 
-bool Topic::init_middleware(const dds::xrce::OBJK_TOPIC_Representation &representation)
-{
-    bool rv = false;
+    Middleware& middleware = participant->get_middleware();
     switch (representation.representation()._d())
     {
         case dds::xrce::REPRESENTATION_BY_REFERENCE:
         {
             const std::string& ref = representation.representation().object_reference();
-            rv = middleware_->create_topic_from_ref(get_raw_id(), participant_->get_raw_id(), ref);
+            created_entity = middleware.create_topic_from_ref(raw_object_id, participant->get_raw_id(), ref);
             break;
         }
         case dds::xrce::REPRESENTATION_AS_XML_STRING:
         {
             const std::string& xml = representation.representation().xml_string_representation();
-            rv = middleware_->create_topic_from_xml(get_raw_id(), participant_->get_raw_id(), xml);
+            created_entity = middleware.create_topic_from_xml(raw_object_id, participant->get_raw_id(), xml);
             break;
         }
         default:
             break;
     }
-    return rv;
+
+    return (created_entity ? new Topic(object_id, participant) : nullptr);
```

Comment:
We should avoid raw pointers. Maybe use smart pointers.

Reply:
Done!